tls/s2n_resume.h

```diff
@@ -99,3 +109,10 @@ extern int s2n_store_to_cache(struct s2n_connection *conn);
  * once we release the session resumption API. */
 int s2n_config_set_initial_ticket_count(struct s2n_config *config, uint8_t num);
 int s2n_connection_add_new_tickets_to_send(struct s2n_connection *conn, uint8_t num);
+
+struct s2n_session_ticket;
+typedef int (*s2n_session_ticket_fn)(struct s2n_connection *conn, struct s2n_session_ticket *ticket);
+int s2n_config_set_session_ticket_cb(struct s2n_config *config, s2n_session_ticket_fn callback, void *ctx);
```

Comment:
Should the context be per-config, or per-connection?

Reply:
Why would the context and the callback be in different places? That seems confusing to me.

Reply:
Nothing forces a 1:1 relationship between callbacks and their contexts. Per-connection contexts are a pretty common use case. It seems unlikely session ticket behavior will vary enough to need different callback methods per-connection, but it might vary enough to need different contexts per-connection. But I don't think we have a specific use case for this context, so it's a judgement call.